### ISSUE TITLE
any simple way to group users?

### ISSUE THREAD

**Issue by ohadcn** [consul/consul#2450]
Is there a simple way to group users? so I can make polls / legislations / proposals / debates for a group (cross geo zones group, like dog owners or religious leaders)
I tried to make an organization - but it won't let me add people to my organization.
this might be related to #2362

**Comment by bertocq** [comment]
Sorry! (missclicked the close button instead of comment :( 🙇 )

At a technical level... Currently there is no User::Group that lets you group users by any filter. What you can see at models Manager/Valuator/Adminstrator/Officer... are more like "profiles". But we haven't encountered yet a use case for a group of users since this app is mainly oriented to citizens and the only "aggrupations" we have the geozones or districts where each of them are censed.

At a product level... I'm not sure I quite follow the scenario/usecase here. Do you mean you want to restrict the actions that users can do (view, vote, comment) over certain resources of the app (polls, proposals...) but.. how's that participatory democracy? I mean we have Universities using consul (that means only certain group of people can get verified and that level allows you actions than non-verified users or guests doesn't have).. but seggregation of registered users 🤔  I guess you have something like a religious/political organization?

If the number of groups you're planning to have is finite, the amount is small (like less than 5 or 10) and the kind of actions/permissions they will have is different and will become a mess to manage that with conditionals in the code ( like `user.group.slug_name == 'dog_owners'`) I would recommend you to consider adding more "profile" models like the mentioned "Managers/Valuators/Administrators". For example given the [`Valuator`](https://github.com/consul/consul/blob/master/app/models/valuator.rb) model you can see we've an `[Abilities::Valuator](https://github.com/consul/consul/blob/master/app/models/abilities/valuator.rb)` class that with the [cancancan](https://github.com/CanCanCommunity/cancancan/) gem usage stores all the permissions that the users with that particulare role have. You could just remove any ability from [Common](https://github.com/consul/consul/blob/master/app/models/abilities/common.rb) users and add new roles with specific ones.

Although.. maybe what you need is to not to enable/disable actions to groups/roles of users in general... but to specific ones 🤔  Like given a Poll "Should we banish from earth dog owners that doesn't clean after their pets" you may just want to disallow or allow the vote for only a "Dog owners" group.. right? If that's the scenario :) you've got a nice and complex feature there! and nothing we've seems like it or may be reused easily to implement it (only the power of Ruby on Rails will).

**Comment by ohadcn** [comment]
How about adding multiple geozones to user? This can also be useful as many of us are leaving in few places (e,g: home / work place)

**Comment by bertocq** [comment]
Yes @ohadcn technically that can be done as well if it suits your needs. But if you're asking if that's a planned feature in consul the answer is no, since right now we're not restricting users actions based on their geozones... they can vote/propose on any district of the city for those reasons (they may sleep in one place but live the entire day in another one). The main problem would be at a product level instead of technical one... how do you gather info about secondary or tertiary geozones? asking the user himself? There needs to be local government laws backing up the usage/restriction of use based on those values, etc..

**Comment by devinbalkind** [comment]
We're using the Participatory Budgeting module and need to limit users to only voting on investments within a specific group. It looks like currently users can vote only vote on one group, but they get to choose which one it is. We need, as administrators, to be able to assign them to the group. These groups are not tied to a specific geography. 

Thank you!
